// 逻辑服务器
syntax = "proto3";
package rpc_server;

service LogicServer 
{
    // 获取玩家收藏
    rpc Get_player_collection(GetPlayerCollectionReq) returns (GetPlayerCollectionRes);
    // 更新玩家收藏
    rpc Update_player_collection(UpdatePlayerCollectionReq) returns (UpdatePlayerCollectionRes);
    // 获取玩家成就
    rpc Get_player_achievements(GetPlayerAchievementsReq) returns (GetPlayerAchievementsRes);
    // 更新玩家成就
    rpc Update_player_achievements(UpdatePlayerAchievementsReq) returns (UpdatePlayerAchievementsRes);
    // 获取玩家任务
    rpc Get_player_tasks(GetPlayerTasksReq) returns (GetPlayerTasksRes);
    // 更新玩家任务
    rpc Update_player_tasks(UpdatePlayerTasksReq) returns (UpdatePlayerTasksRes);
    // 添加物品
    rpc Add_item(AddItemReq) returns (AddItemRes);
    // 使用物品
    rpc Use_item(UseItemReq) returns (UseItemRes);
    // 保存对局结果
    rpc Save_battle_result(SaveBattleResultReq) returns (SaveBattleResultRes);
}

// 获取玩家收藏请求
message GetPlayerCollectionReq
{
    string player_id = 1;
}
// 获取玩家收藏响应
message GetPlayerCollectionRes
{
    string player_id = 1;
    repeated string collection = 2; // 玩家收藏的卡牌ID列表
}

// 更新玩家收藏请求
message UpdatePlayerCollectionReq
{
    string player_id = 1;
    repeated string collection = 2; // 更新后的玩家收藏的卡牌ID列表
}
// 更新玩家收藏响应
message UpdatePlayerCollectionRes
{
    bool success = 1;
    string message = 2;
}

// 获取玩家成就请求
message GetPlayerAchievementsReq
{
    string player_id = 1;
}
// 获取玩家成就响应
message GetPlayerAchievementsRes
{
    string player_id = 1;
    repeated string achievements = 2; // 玩家成就ID列表
}

// 更新玩家成就请求
message UpdatePlayerAchievementsReq
{
    string player_id = 1;
    repeated string achievements = 2; // 更新后的玩家成就ID列表
}
// 更新玩家成就响应
message UpdatePlayerAchievementsRes
{
    bool success = 1;
    string message = 2;
}

// 获取玩家任务请求
message GetPlayerTasksReq
{
    string player_id = 1;
}
// 获取玩家任务响应
message GetPlayerTasksRes
{
    string player_id = 1;
    repeated string tasks = 2; // 玩家任务ID列表
}

// 更新玩家任务请求
message UpdatePlayerTasksReq
{
    string player_id = 1;
    repeated string tasks = 2; // 更新后的玩家任务ID列表
}
// 更新玩家任务响应
message UpdatePlayerTasksRes
{
    bool success = 1;
    string message = 2;
}

// 添加物品请求
message AddItemReq
{
    string player_id = 1;
    string item_id = 2; // 要添加的物品ID
}
// 添加物品响应
message AddItemRes
{
    bool success = 1;
    string message = 2;
}

// 使用物品请求
message UseItemReq
{
    string player_id = 1;
    string item_id = 2; // 要使用的物品ID
}
// 使用物品响应
message UseItemRes
{
    bool success = 1;
    string message = 2;
}

// 保存对局结果请求
message SaveBattleResultReq
{
    string player_id = 1;
    string opponent_id = 2;
    bool is_victory = 3; // 是否胜利
    string battle_data = 4; // 对局详细数据
}
// 保存对局结果响应
message SaveBattleResultRes
{
    bool success = 1;
    string message = 2;
}

